lib/es6/toolset.js

```diff
@@ -60,8 +60,14 @@ Toolset.prototype.initializePosix = function (install) {
         this.cCompilerPath = "gcc";
     }
 
+    if( this.options.generator ){
+        if (install) {
+            this.log.info("TOOL", "Using " + this.options.generator + " generator, as specified from commandline.");
+        }
+        this.generator = this.options.generator;
+    }
     // 2: Generator
-    if (environment.isOSX) {
+    else if (environment.isOSX) {
```

Comment:
I wouldn't put generator override code at this point. Let toolset find its defaults and then, after initialize() around line 34, we should override the default value by logging out the exact information about which default generator got changed into which one.